src/EditorFeatures/TestUtilities/Threading/StaTaskScheduler.cs

```diff
@@ -18,6 +18,9 @@ public sealed class StaTaskScheduler : IDisposable
 
         public bool IsRunningInScheduler => StaThread.ManagedThreadId == Thread.CurrentThread.ManagedThreadId;
 
+        // An int to allow use of Interlocked.CompareExchange
+        private static int s_SubscribedToAppDomainUnload = 0;
```

Comment:
💡 Just make the constructor private and this can be removed.

Reply:
I didn't want it to trigger unless the regular constructor was actually used; maybe that was being too paranoid?

Reply:
(static constructors are often evil in themselves, ones that are using AppDomain.CurrentDomain in a behavior-changing way seemed...unethical.)

Reply:
The regular constructor is already called by the static constructor. The only use is to initialize a singleton property.